### DISCUSSION TITLE
question on behaviour of setSystemTime in threaded vs non-threaded

### DISCUSSION THREAD

**Discussion by capaj** [vitest-dev/vitest#1836] [Q&A]
Is setSystemTime behaviour different when user is running threaded vs non threaded?
Like when running non threaded does it persist across tests or does vitest call `useRealTimers` in between tests?

**Comment by sheremet-va** [comment]
it persists across tests

**Reply by capaj** [reply]
even when threaded?

**Reply by sheremet-va** [reply]
only with `--no-threads`

**Reply by capaj** [reply]
Would be worth documenting this. I'll open a PR

**Reply by sheremet-va** [reply]
> Would be worth documenting this. I'll open a PR

Not sure if it's worth it. Every test in `--no-threads` is executed in a single thread with a shared global scope. I think it's already documented. If not, we should document _that_.
